File: VB/CustomFilterPopupSample/Form1.vb
Imports DevExpress.XtraEditors
Imports System
Imports System.Windows.Forms

Namespace CustomFilterPopupSample
	Partial Public Class CustomFilterPopupForm
		Inherits XtraForm

		Public Sub New()
			InitializeComponent()
		End Sub
		Private Sub CustomFilterPopupForm_Load(ByVal sender As Object, ByVal e As EventArgs) Handles Me.Load
			CustomizePivotGrid(pivotGrid)
			pivotGrid.BestFit()
		End Sub

		Private Sub cbEnableCustomFilterPopup_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cbEnableCustomFilterPopup.CheckedChanged
			If (TryCast(sender, CheckBox)).Checked Then
				productNameFilterRanges.AddEvents()
				dateFilterRanges.AddEvents()
			Else
				productNameFilterRanges.Detach()
				dateFilterRanges.Detach()
			End If
		End Sub
	End Class
End Namespace

File: CS/CustomFilterPopupSample/Form1.cs
using DevExpress.XtraEditors;
using System;
using System.Windows.Forms;

namespace CustomFilterPopupSample
{
    public partial class CustomFilterPopupForm : XtraForm {
        public CustomFilterPopupForm() {
            InitializeComponent();
        }
        private void CustomFilterPopupForm_Load(object sender, EventArgs e) {
            CustomizePivotGrid(pivotGrid);
            pivotGrid.BestFit();
        }

        private void cbEnableCustomFilterPopup_CheckedChanged(object sender, EventArgs e) {
            if ((sender as CheckBox).Checked) {
                productNameFilterRanges.AddEvents();
                dateFilterRanges.AddEvents();
            } else {
                productNameFilterRanges.Detach();
                dateFilterRanges.Detach();
            }
        }
    }
}
